File: CS/WindowsApplication1/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraEditors.Repository;
using DevExpress.XtraEditors;

namespace WindowsApplication1
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
            popupContainerEdit1.Text = ValuesConverter.TrackBarRangeToString(new TrackBarRange(5, 10));
        }


        private void OnQueryResultValue(object sender, DevExpress.XtraEditors.Controls.QueryResultValueEventArgs e)
        {
            e.Value = ValuesConverter.TrackBarRangeToString(rangeTrackBarControl1.Value);
        }

        private void OnQueryPopup(object sender, CancelEventArgs e)
        {
            TrackBarRange value = ValuesConverter.StringToTrackBarRange((sender as PopupContainerEdit).Text);
            if (value.IsEmpty)
                e.Cancel = true;
            else
                ApplyValue(value);
        }
        private void ApplyValue(TrackBarRange value)
        {
            RepositoryItemRangeTrackBar properties = rangeTrackBarControl1.Properties;
            properties.Minimum = Math.Min(properties.Minimum, value.Minimum);
            properties.Maximum = Math.Max(properties.Maximum, value.Maximum);
            rangeTrackBarControl1.Value = value;
        }

        private void simpleButton1_Click(object sender, EventArgs e)
        {
            popupContainerEdit1.ClosePopup();
        }
    }
}

File: VB/WindowsApplication1/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraEditors.Repository
Imports DevExpress.XtraEditors

Namespace WindowsApplication1
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
			popupContainerEdit1.Text = ValuesConverter.TrackBarRangeToString(New TrackBarRange(5, 10))
		End Sub


		Private Sub OnQueryResultValue(ByVal sender As Object, ByVal e As DevExpress.XtraEditors.Controls.QueryResultValueEventArgs) Handles popupContainerEdit1.Properties.QueryResultValue
			e.Value = ValuesConverter.TrackBarRangeToString(rangeTrackBarControl1.Value)
		End Sub

		Private Sub OnQueryPopup(ByVal sender As Object, ByVal e As CancelEventArgs) Handles popupContainerEdit1.QueryPopUp
			Dim value As TrackBarRange = ValuesConverter.StringToTrackBarRange((TryCast(sender, PopupContainerEdit)).Text)
			If value.IsEmpty Then
				e.Cancel = True
			Else
				ApplyValue(value)
			End If
		End Sub
		Private Sub ApplyValue(ByVal value As TrackBarRange)
			Dim properties As RepositoryItemRangeTrackBar = rangeTrackBarControl1.Properties
			properties.Minimum = Math.Min(properties.Minimum, value.Minimum)
			properties.Maximum = Math.Max(properties.Maximum, value.Maximum)
			rangeTrackBarControl1.Value = value
		End Sub

		Private Sub simpleButton1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles simpleButton1.Click
			popupContainerEdit1.ClosePopup()
		End Sub
	End Class
End Namespace